From 0976d5303024910aa2ccd02d303aa648dca06f7d Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Fri, 21 Aug 2020 11:38:05 +0800
Subject: [PATCH] drm: Remove hdr_metadata_changed

For compatibility with GKI, hdr_metadata_changed should be removed.

Fixes: 2beb3925b795 ("drm: bridge: dw-hdmi: support Dynamic Range and Mastering Infoframe")
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
Change-Id: Ia1981a44507bbd58e5677a9ebfc0053afd228d19
---
 drivers/gpu/drm/drm_atomic.c        | 1 -
 drivers/gpu/drm/drm_atomic_helper.c | 1 -
 include/drm/drm_connector.h         | 2 --
 3 files changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/drm_atomic.c b/drivers/gpu/drm/drm_atomic.c
index 93aa30671ea7..f64453298c94 100644
--- a/drivers/gpu/drm/drm_atomic.c
+++ b/drivers/gpu/drm/drm_atomic.c
@@ -1406,7 +1406,6 @@ static int drm_atomic_connector_set_property(struct drm_connector *connector,
 				val,
 				sizeof(struct hdr_output_metadata), -1,
 				&replaced);
-		state->hdr_metadata_changed |= replaced;
 		return ret;
 	} else if (property == config->aspect_ratio_property) {
 		state->picture_aspect_ratio = val;
diff --git a/drivers/gpu/drm/drm_atomic_helper.c b/drivers/gpu/drm/drm_atomic_helper.c
index 9b4db909db5d..9274ac77ed25 100644
--- a/drivers/gpu/drm/drm_atomic_helper.c
+++ b/drivers/gpu/drm/drm_atomic_helper.c
@@ -3843,7 +3843,6 @@ __drm_atomic_helper_connector_duplicate_state(struct drm_connector *connector,
 	if (state->hdr_output_metadata)
 		drm_property_blob_get(state->hdr_output_metadata);
 
-	state->hdr_metadata_changed = false;
 	/* Don't copy over a writeback job, they are used only once */
 	state->writeback_job = NULL;
 }
diff --git a/include/drm/drm_connector.h b/include/drm/drm_connector.h
index 6a84723eae4d..281ea0fc59c4 100644
--- a/include/drm/drm_connector.h
+++ b/include/drm/drm_connector.h
@@ -546,8 +546,6 @@ struct drm_connector_state {
 	struct drm_property_blob *hdr_output_metadata;
 
 	struct drm_property_blob *hdr_panel_blob_ptr;
-
-	bool hdr_metadata_changed : 1;
 };
 
 /**
-- 
2.35.3

